src/main/kotlin/com/github/gradle/node/NodeExtension.kt

```diff
@@ -59,6 +59,14 @@ open class NodeExtension(project: Project) {
      */
     val distBaseUrl = project.objects.property<String>()
 
+    /**
+     * Specifies whether it is acceptable to communicate with the Node.js repository over an insecure HTTP connection.
+     * Only used if download is true
+     * Change it to true if you use a mirror that uses HTTP rather than HTTPS
+     * Or set to null if you want to use Gradle's default behaviour.
+     */
+    val allowInsecureProtocol = project.objects.property<Boolean>().convention(null)
```

Comment:
Instead of defaulting this to `null` I think we might be better off just make setting it conditional depending on the gradle version (and maybe the value)

e.g. if on gradle 7.0 or the value is set to `false`.

Reply:
Sure.  I think it's desirable to be able to set it on Gradle >= 6.0 because it triggers a warning between >=6.0 and <7.0.

Reply:
It looks like it was added in 6.0 so we'd be breaking backwards compatibility otherwise and #159 is probably not happening just yet